plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -83,51 +93,48 @@ public ConnectorTableMetadata getTableMetadata()
     }
 
     @Override
-    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
     {
-        if (commitInfoEntries.isEmpty()) {
-            return new FixedPageSource(ImmutableList.of());
-        }
-        return new FixedPageSource(buildPages(session));
-    }
-
-    private List<Page> buildPages(ConnectorSession session)
-    {
-        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
         TimeZoneKey timeZoneKey = session.getTimeZoneKey();
+        Iterable<List<Object>> records = () ->
+                stream(commitInfoEntries)
+                        .map(commitInfoEntry -> getRecord(commitInfoEntry, timeZoneKey))
+                        .iterator();
```

Comment:
Rather than going back and forth to `Stream` you can just use `Iterables.transform` or `Iterators.transform`